src/osd/PGLog.h

```diff
@@ -1410,11 +1410,12 @@ struct PGLog : DoutPrefixProvider {
 	for (list<pg_log_entry_t>::reverse_iterator i = log.log.rbegin();
 	     i != log.log.rend();
 	     ++i) {
-	  if (!debug_verify_stored_missing && i->version <= info.last_complete) break;
 	  if (i->soid > info.last_backfill)
 	    continue;
 	  if (i->is_error())
 	    continue;
+    if (!i->is_delete())
+        missing.merge(*i);
```

Comment:
My understanding is that read_log_and_missing() is only invoked this way: OSD::load_pgs() --> PG::read_state() --> PGLog::read_log_and_missing().
So missing.merge() here is just invoked when an OSD boots and more complicated during PG peering has not been addressed yet, right?

Reply:
Sorry for replying so late，Yes, you are right.